apps/mitt-konto/src/MittKonto/Main/Views/UserView.purs

```diff
@@ -147,6 +161,102 @@ userView router { state: { now, news } } logger profileComponent user = React.fr
             , target: "_blank"
             }
 
+    mailinglistView =
+      componentBlock "Mina nyhetsbrev:" $ [ componentBlockContent $ contents ]
+      where
+        updateResultMessage :: JSX
+        updateResultMessage = case newslettersUpdated of
+            Types.NotUpdated -> mempty
+            Types.Updating -> mempty
+            Types.Updated -> DOM.p_ [ DOM.text "Uppdateringen av prenumerationen på nyhetsbrevet lyckades." ]
+            Types.UpdateFailed -> DOM.p_ [ DOM.text "Något gick fel med uppdateringen av prenumerationen på nyhetsbrevet." ]
+
+        contents :: JSX
+        contents =
+          DOM.div
+            { className: "mitt-konto--newsletters"
+            , children: case activeUserNewsletters of
+                Nothing -> [ loadingSpinner ]
+                Just allNewsletters ->
+                  [ checkboxes allNewsletters
+                  , Elements.break
+                  , acceptChangesButton
+                  ]
+            }
+
+        newsletterCheckbox :: (NewsletterSubscription -> Effect Unit) -> NewsletterSubscription -> JSX
+        newsletterCheckbox updateSubs newsletter =
+          DOM.dd_
+            [ DOM.label_
+                [ DOM.input
+                    { type: "checkbox"
+                    , checked: newsletter.subscribed
+                    , onChange: handler targetValue \_ -> updateSubs (newsletter {subscribed = not newsletter.subscribed} )
+                    }
+                , DOM.text $ newsletter.listName ]]
+
+        replaceMatching :: forall f a b. Functor f => Eq b => (a -> b) -> a -> f a -> f a
+        replaceMatching fn replacement arr = map (\existing -> if fn existing == fn replacement then replacement else existing) arr
+
+        paperSection :: Newsletter -> (Newsletter -> Effect Unit) -> Array JSX
+        paperSection paperNewsletters updateNewsletter =
+          DOM.dt_
+            [ DOM.text $ Paper.paperName paperNewsletters.paper
+            , legalLink paperNewsletters.paper
+            ]
+          : concatMap
+              (uncurry
+                (\categoryId newsletters ->
+                  map (newsletterCheckbox
+                        (\replacement ->
+                          let subscriptions = replaceMatching _.id replacement newsletters
+                              newSubs = Map.insert categoryId subscriptions paperNewsletters.subscriptions
+                          in updateNewsletter $ paperNewsletters {subscriptions = newSubs})) newsletters))
+              (Map.toUnfoldable paperNewsletters.subscriptions)
+
+        checkboxes :: Array Newsletter -> JSX
+        checkboxes allNewsletters =
+          DOM.dl_ $
+            concatMap
+              (\paper -> paperSection paper (\replacement ->
+               let replaced = replaceMatching _.listId replacement allNewsletters
+               in launchAff_ $ updateNewsletters replaced
+              ))
+            allNewsletters
+
+        link :: String -> JSX
```

Comment:
This is used only by `legalLink` right? I'd restrict it to that scope, it's a bit too generically named for this section, with a fixed text too.